src/control/lib/cache/cache.go

```diff
@@ -0,0 +1,225 @@
+//
+// (C) Copyright 2023 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package cache
```

Comment:
is https://pkg.go.dev/istio.io/pkg/cache unsuitable? do we need to roll our own?

Reply:
The cache lib ends up being pretty simple, especially with some changes suggested by Mike. I think rolling our own makes sense here, as the functionality in the linked package is also a little different from what we're looking for.